include/mapped_soa.h

```diff
@@ -85,6 +85,27 @@ class MappedSoA {
     key_to_index_.erase(key);
   }
 
+  void copy(size_t index0, size_t index1) {
+    const size_t size_val = size();
+    if (size_val <= index0 || size_val <= index1) {
+      assert(false);
+
+      return;
+    }
+
+    if (index0 == index1) {
+      return;
+    }
+
+    const KeyType key0 = index_to_key_[index0];
+    const KeyType key1 = index_to_key_[index1];
+
+    key_to_index_[key1] = key_to_index_[key0];
```

Comment:
Do not copy the index. The index refers to the index of the key's values in the array.